src/ch/ehi/umleditor/application/TreeTransferHandler.java

```diff
@@ -0,0 +1,217 @@
+import ch.ehi.interlis.domainsandconstants.linetypes.LineFormTypeDef;
+import ch.ehi.interlis.functions.FunctionDef;
+import ch.ehi.interlis.graphicdescriptions.GraphicParameterDef;
+import ch.ehi.interlis.metaobjects.MetaDataUseDef;
+import ch.ehi.interlis.modeltopicclass.ClassDef;
+import ch.ehi.interlis.modeltopicclass.INTERLIS2Def;
+import ch.ehi.interlis.modeltopicclass.ModelDef;
+import ch.ehi.interlis.modeltopicclass.TopicDef;
+import ch.ehi.interlis.units.UnitDef;
+import ch.ehi.interlis.views.ViewDef;
+import ch.ehi.uml1_4.foundation.core.Element;
+import ch.ehi.uml1_4.foundation.core.ModelElement;
+import ch.ehi.uml1_4.foundation.core.Namespace;
+import ch.ehi.uml1_4.implementation.UmlPackage;
+
+public class TreeTransferHandler extends TransferHandler {
+	private static final long serialVersionUID = -780714214950735708L;
+	private static java.util.ResourceBundle resTreeTransferHandler = java.util.ResourceBundle
+			.getBundle("ch/ehi/umleditor/application/resources/TreeTransferHandler");
+	DataFlavor nodesFlavor;
+    DataFlavor[] flavors = new DataFlavor[1];
+    Element[] nodesToRemove;
+     public TreeTransferHandler() {
+        try {
+            String mimeType = DataFlavor.javaJVMLocalObjectMimeType +
+                              ";class=\"" +
+                Namespace.class.getName() +
+                              "\"";
+            nodesFlavor = new DataFlavor(mimeType);
+            flavors[0] = nodesFlavor;
+        } catch(ClassNotFoundException e) {
+            System.out.println("ClassNotFound: " + e.getMessage());
+        }
+    }
+     /**
+     *  Drag only nodes of Jtree
+     */
+    public boolean canImport(TransferHandler.TransferSupport support) {
+        if(!support.isDrop()) {
+            return false;
+        }
+        support.setShowDropLocation(true);
+        if(!support.isDataFlavorSupported(nodesFlavor)) {
+            return false;
+        }
+        return true;
+    }
+     protected Transferable createTransferable(JComponent c) {
+        JTree tree = (JTree)c;
+        TreePath[] paths = tree.getSelectionPaths();
+        if(paths != null) {
+            // Make up a node array of copies for transfer and
+            // another for/of the nodes that will be removed in
+            // exportDone after a successful drop.
+            List<Element> copies = new ArrayList<Element>();
+            List<Element> toRemove = new ArrayList<Element>();
+            Element node = (Element)paths[0].getLastPathComponent();
+            Element copy = node;
+            copies.add(copy);
+            toRemove.add(node);
+            
+            Element[] nodes = copies.toArray(new Element[copies.size()]);
+            nodesToRemove = toRemove.toArray(new Element[toRemove.size()]);
+            return new NodesTransferable(nodes);
+        }
+        return null;
+    }
+
+     public int getSourceActions(JComponent c) {
+        return COPY_OR_MOVE;

```

Comment:
COPY will not work, if the node contains references to elements outside itself or its children (such as an attribute referencing a domain). Change to MOVE.